cheatsheets/Protect_FileUpload_Against_Malicious_File.md

```diff
@@ -1,597 +1,153 @@
-# Introduction
+# File Upload Cheat Sheet
 
-This article propose a way to protect a file upload feature against submission of file containing malicious code.
+## Introduction
 
-# Context
+File upload is becoming more and more an essential part of any application, where the user is able to upload their photo, their CV, or a video showcasing a project they are working on. It is evident that applications should be able to handle file uploads based on the business requirement. In return, the application should be able to fend off bogus, and malicious files in a way to keep the application and the users safe.
 
-Into web applications, when we expect upload of working documents from users, we can expose the application to submission of documents that we can categorize as *malicious*.
+In short, the following principles should be followed to reach a secure file upload implementation:
 
-We use the term "malicious" here to refer to documents that embed *malicious code* that will be executed when another user (admin, back office operator...) will open the document with the associated application reader.
+- **Whitelist allowed extensions. Only allow safe and critical extensions for business functionality**
+  - **Ensure that [input validation](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Input_Validation_Cheat_Sheet.md#validating-free-form-unicode-text) is applied before validating the extensions.**
+- **Allowed users only can upload files**
+- **Validate the file type, don't trust the [Content-Type header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)  as it can be spoofed**
```

Comment:
In the [File Upload Protection](#file-upload-protection) there no mention of validating the file type/content - you wrote about validating extensions and Content-Type but nothing about actual content of a file (we can recommend here libraries/frameworks that do that)

Reply:
Hmm, I'll add to it. I can't sustain that though in the document. It's very difficult. There is a whole wide of file types to cover, lots of languages/frameworks. I'd prefer just putting the note, and if it were to be done, they'd need a person with enough security expertise.